FirebaseUI/Database/API/FirebaseArray.h

```diff
@@ -39,43 +49,36 @@ NS_ASSUME_NONNULL_BEGIN
  * The delegate object that array changes are surfaced to, which conforms to the
  * [FirebaseArrayDelegate Protocol](FirebaseArrayDelegate).
  */
-@property(weak, nonatomic) id<FirebaseArrayDelegate> delegate;
+@property(weak, nonatomic, nullable) id<FirebaseArrayDelegate> delegate;
 
 /**
  * The query on a Firebase reference that provides data to populate the instance of FirebaseArray.
  */
-@property(strong, nonatomic) FIRDatabaseQuery *query;
+@property(strong, nonatomic) id<FIRDataObservable> query;
 
 /**
- * The delegate object that array changes are surfaced to.
+ * The number of objects in the FirebaseArray.
  */
-@property(strong, nonatomic) NSMutableArray<FIRDataSnapshot *> * snapshots;
-
-#pragma mark -
-#pragma mark Initializer methods
+@property(nonatomic, readonly) NSUInteger count;
 
 /**
- * Intitalizes FirebaseArray with a standard Firebase reference.
- * @param ref The Firebase reference which provides data to FirebaseArray
- * @return The instance of FirebaseArray
+ * The items currently in the FirebaseArray.
  */
-- (instancetype)initWithRef:(FIRDatabaseReference *)ref;
+@property(nonatomic, readonly, copy) NSArray *items;
```

Comment:
I don't believe copy copies on read, only write. 

Regarding the style, but in the absence of a known community opinion we should prefer existing style guides. If we diverge from that, we should note that we are, and document our own style variations. 

Regardless of what we choose, we should be consistent - so either change this one to match the order in https://github.com/firebase/FirebaseUI-iOS/pull/108/files#diff-13a906499860a93cf34277f908dfb0b7R37 or vice versa.